src/porepy/grids/mdg_generation.py

```diff
@@ -0,0 +1,295 @@
+    y_space = None
+    z_space = None
+
+    xmin = omega.bounding_box["xmin"]
+    xmax = omega.bounding_box["xmax"]
+    ymin = omega.bounding_box["ymin"]
+    ymax = omega.bounding_box["ymax"]
+
+    h_size = mesh_arguments["mesh_size"]
+    n_x = round((xmax - xmin) / h_size) + 1
+    n_y = round((ymax - ymin) / h_size) + 1
+
+    x_space = np.linspace(xmin, xmax, num=n_x)
+    y_space = np.linspace(ymin, ymax, num=n_y)
+
+    # Avoid duplicity in kwargs
+    x_arg = kwargs.get("x", None)
+    if isinstance(x_arg, x_space.__class__):
+        x_space = x_arg
+        kwargs.pop("x")
+    y_arg = kwargs.get("y", None)
+    if isinstance(y_arg, y_space.__class__):
+        y_space = y_arg
+        kwargs.pop("y")
+
+    if omega.dim == 3:
+        zmin = omega.bounding_box["zmin"]
+        zmax = omega.bounding_box["zmax"]
+        n_z = round((zmax - zmin) / h_size) + 1
+        z_space = np.linspace(zmin, zmax, num=n_z)
+
+        z_arg = kwargs.get("z", None)
+        if isinstance(z_arg, z_space.__class__):
+            z_space = z_arg
+            kwargs.pop("z")
+
+    # remove duplicate keys
+    [kwargs.pop(item[0]) for item in mesh_arguments.items() if item[0] in kwargs]
+
+    return (x_space, y_space, z_space, kwargs)
+
+
+def create_mdg(
+    grid_type: Literal["simplex", "cartesian", "tensor_grid"],
+    mesh_arguments: dict,
+    fracture_network: FractureNetwork,
+    **kwargs,
+) -> pp.MixedDimensionalGrid:
+    """Creates a mixed dimensional grid.
+
+    Examples:
+
+        .. code:: python3
+
+            import porepy as pp
+            import numpy as np
+
+            # Set a domain
+            domain = pp.Domain({"xmin": 0, "xmax": 1, "ymin": 0, "ymax": 1})
+
+            # Generate a fracture network
+            frac_1 = pp.LineFracture(np.array([[0.0, 2.0], [0.0, 0.0]]))
+            frac_2 = pp.LineFracture(np.array([[1.0, 1.0], [0.0, 1.0]]))
+            fractures = [frac_1, frac_2]
+            fracture_network = pp.create_fracture_network(fractures, domain)
+
+            # Generate a mixed-dimensional grid (MDG)
+            mesh_args = {"mesh_size": 0.1}
+            mdg = pp.create_mdg("simplex",mesh_args,fracture_network)
+            pp.Exporter(mdg, "mdg_with_fractures").write_vtu()
+
+    Raises:
+        - ValueError: If ``fracture_network.domain.dim`` is not 2 or 3.
+            If
+        - TypeError: Mandatory arguments types inconsistent (see type hints in signature).
+
+    Parameters:
+        grid_type: Type of the resulting grid
+            instance of Literal["simplex", "cartesian", "tensor_grid"].
+        mesh_arguments: ... .
+        domain: Domain specfication. An instance of
+            :class:`~porepy.geometry.domain.Domain`.
+        fracture_network: fracture network specification. Instance of
+            :class:`~porepy.fracs.fracture_network_2d` or
+            :class:`~porepy.fracs.fracture_network_3d`.
+        **kwargs: Extra arguments associated with each grid_type.
```

Comment:
Here, I think we should add for each type grid and dimension combination, what are the admissible **kwargs (or at least outsource to the relevant method/class). Otherwise, it will be quite difficult to keep track of them.

Reply:
Based on our core developer meeting. Some arguments were translated to meshing_agrs. The documentation of the admissible arguments is now provided.